nova/objects/flavor.py

```diff
@@ -632,10 +641,10 @@ def _flavor_get_all_from_db(context, inactive, filters, sort_key, sort_dir,
 
     all_flavors = [_dict_with_extra_specs(i) for i in query.all()]
 
-    aliased_flavors = [_aliased_flavor_dict(f)
-                       for f in all_flavors
-                       if f['extra_specs'].get('catalog:alias')
-                           and f['is_public']]
+    aliased_flavors = []
+    for f in all_flavors:
+        if f['extra_specs'].get('catalog:alias') and f['is_public']:
```

Comment:
Since this is a conditional check it seems more natural to me to check for `'catalog:alias' in f['extra_specs']`.

Reply:
Heh: https://github.com/sapcc/nova/pull/189/files/a1704d976b2934110aa211a290dd32991fa4a8db#diff-cb7a1c0ee6867c2868aacfaf2ef9c8d4329347af02664a6f7d26057af33fc0ca 🙂 

It checks two things at once, the presence of the key, and it being non-empty/-none/etc. It's shorter but -- apparently -- harder to read. I think I'd like to keep it this way.